lib/gis/mapset_msc.c

```diff
@@ -183,8 +183,9 @@ int G_make_mapset_object_group_basedir(const char *type, const char *basedir)
 int make_mapset_element_impl(const char *p_path, const char *p_element,
                              bool race_ok)
 {
-    char path[GPATH_MAX], *p;
+    char path[GPATH_MAX + 1], *p;
     const char *element;
+    path[GPATH_MAX] = 0;
```

Comment:
Although I do not think here not-null terminated p_path can reach this function, it wouldn't harm either.

Please change from 0 to '\0' to make your intent clearer.